crates/ruff_linter/src/rules/pydocstyle/rules/backslashes.rs

```diff
@@ -69,8 +69,35 @@ pub(crate) fn backslashes(checker: &mut Checker, docstring: &Docstring) {
     // Docstring contains at least one backslash.
     let body = docstring.body();
     let bytes = body.as_bytes();
+    let mut backslash_index = 0;
+    let escaped_docstring_backslashes_pattern = b"\"\\\"\\\"";
```

Comment:
We likely also need to handle single quotes here (i.e., escaped single quotes within single-quote docstrings).

Reply:
But D301 is based on double quotes, do we need to cover single-quote docstring here?

Reply:
I haven't verified this myself but the way I read the code is that docstrings are extracted from any string literal 

https://github.com/astral-sh/ruff/blob/7d16f831c4af4c78344c8e9a1bb8259fb425d671/crates/ruff_linter/src/docstrings/extraction.rs#L6-L15